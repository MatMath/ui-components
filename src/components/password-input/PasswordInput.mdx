---
name: Password Input
menu: Components
---

import { Playground, Props } from 'docz';
import { PasswordInput } from './PasswordInput';

# Password Input

A Password input component

## Properties

<Props of={PasswordInput} />

## Usage

### Basic usage

Examples below show usage of the Password Input

<Playground
  style={{
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: 400,
    flex: 1
  }}
>
  <PasswordInput
    hidePasswordTooltipText='Hide'
    showPasswordTooltipText='Reveal'
    label='Password'
    placeholder='Test'
  />
  <PasswordInput
    hidePasswordTooltipText='Hide'
    showPasswordTooltipText='Reveal'
    label='Password'
    placeholder='Test'
    onPasswordVisibilityChanged={nextState => console.log(nextState)}
  />
  <PasswordInput
    hidePasswordTooltipText='Hide'
    showPasswordTooltipText='Reveal'
  />
  <PasswordInput
    hidePasswordTooltipText='Hide'
    showPasswordTooltipText='Reveal'
    disabled
  />
  <PasswordInput
    hidePasswordTooltipText='Hide'
    showPasswordTooltipText='Reveal'
    feedbackType='warning'
    feedbackText='This password contains invalid characters'
  />
  <PasswordInput
    hidePasswordTooltipText='Hide'
    showPasswordTooltipText='Reveal'
    feedbackType='error'
    feedbackText='Wrong password'
  />
</Playground>
