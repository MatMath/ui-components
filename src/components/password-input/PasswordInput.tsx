import * as React from 'react';

import { getClassNames } from '@utility/cssUtils';
import { HideIcon, RevealIcon } from '@atoms/icons';
import { midGreen00 } from '@colors';
import { Tooltip } from '@components';

import styles from './PasswordInput.module.scss';

type PasswordInputFeedbackType = 'error' | 'warning';

type _ForbiddenProps = 'size' | 'prefix' | 'type';

export interface PasswordInputProps
  extends Omit<React.InputHTMLAttributes<HTMLInputElement>, _ForbiddenProps> {
  /**
   * Some text to display under the input when triggering a feedback type
   */
  feedbackText?: string;
  /**
   * The feedback nature of the input
   */
  feedbackType?: PasswordInputFeedbackType;
  /**
   * Make this field take 100% of the width of it's container
   * @default false
   */
  fluid?: boolean;
  /**
   * The text that is displayed on hover on the hide icon when the
   * password is shown
   */
  hidePasswordTooltipText: string;
  /**
   * The label used to describe the field, it's displayed on top.
   * Note that the text will be uppercase
   */
  label?: string;
  /**
   * A function that is triggered when the visibility
   * of the password is changed.
   */
  onPasswordVisibilityChanged?: (visible: boolean) => void;
  /**
   * The text that is displayed on hover on the reveal icon when the
   * password is hidden
   */
  showPasswordTooltipText: string;
}

export const PasswordInput = (props: PasswordInputProps): JSX.Element => {
  const {
    feedbackText,
    feedbackType,
    fluid = false,
    hidePasswordTooltipText,
    label,
    onPasswordVisibilityChanged,
    showPasswordTooltipText,
    ...prop
  } = props;

  const [showPassword, setShowPassword] = React.useState(false);
  const inputRef = React.createRef<HTMLInputElement>();

  const handleShowPassword = () => {
    if (onPasswordVisibilityChanged) {
      onPasswordVisibilityChanged(!showPassword);
    }
    setShowPassword(prevState => !prevState);
    inputRef.current && inputRef.current.focus();
  };

  const styledClassName = (feedbackType && styles[feedbackType]) || '';

  const fluidClassName = (fluid && styles.fluid) || '';

  const RevealOrHideIcon = showPassword ? HideIcon : RevealIcon;

  return (
    <div className={getClassNames(styles.root, fluidClassName)}>
      {label && <span className={styles.label}>{label}</span>}
      <div className={getClassNames(styles.container, fluidClassName)}>
        <input
          ref={inputRef}
          {...prop}
          type={showPassword ? 'text' : 'password'}
          className={getClassNames(
            styles.input,
            styledClassName,
            fluidClassName
          )}
        />
        <button
          className={getClassNames(styles.icon, fluidClassName)}
          onClick={handleShowPassword}
        >
          {prop.disabled ? (
            <RevealOrHideIcon disabled={prop.disabled} />
          ) : (
            <Tooltip
              placement='left'
              title={
                showPassword ? hidePasswordTooltipText : showPasswordTooltipText
              }
            >
              <RevealOrHideIcon hoverColor={midGreen00} />
            </Tooltip>
          )}
        </button>
      </div>
      {feedbackText && feedbackType && (
        <span className={getClassNames(styles.feedback, styledClassName)}>
          {feedbackText}
        </span>
      )}
    </div>
  );
};
