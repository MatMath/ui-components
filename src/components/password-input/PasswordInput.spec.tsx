import * as React from 'react';
import { mount } from 'enzyme';
import { PasswordInput } from './PasswordInput';

describe('<Button>', () => {
  describe('Global render', () => {
    it('should render default', () => {
      const passwordInput = mount(<PasswordInput />);
      expect(passwordInput).toMatchSnapshot();
    });

    it('should render warning state without feedback text', () => {
      const passwordInput = mount(<PasswordInput feedbackType='warning' />);
      expect(passwordInput).toMatchSnapshot();
    });

    it('should render warning state with feedback text', () => {
      const passwordInput = mount(
        <PasswordInput feedbackType='warning' feedbackText='Some feedback' />
      );
      expect(passwordInput).toMatchSnapshot();
    });

    it('should render error state without feedback text', () => {
      const passwordInput = mount(<PasswordInput feedbackType='error' />);
      expect(passwordInput).toMatchSnapshot();
    });

    it('should render error state with feedback text', () => {
      const passwordInput = mount(
        <PasswordInput feedbackType='error' feedbackText='Some feedback' />
      );
      expect(passwordInput).toMatchSnapshot();
    });

    it('should render with password visible', () => {
      const passwordInput = mount(
        <PasswordInput
          showPassword={true}
          value='SomeRandomPassword'
          onChange={() => {}}
        />
      );
      expect(passwordInput).toMatchSnapshot();
    });

    it('should render with weakest password strength indicator visible', () => {
      const passwordInput = mount(
        <PasswordInput
          showPasswordStrength={true}
          passwordStrengthProps={{ score: 0 }}
        />
      );
      expect(passwordInput).toMatchSnapshot();
    });
  });
});
