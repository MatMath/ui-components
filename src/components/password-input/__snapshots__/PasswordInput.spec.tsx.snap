// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordInput /> should render default 1`] = `
<Component
  hidePasswordTooltipText="Hide"
  label="Password"
  showPasswordTooltipText="Reveal"
>
  <div
    className="root"
  >
    <span
      className="label"
    >
      Password
    </span>
    <div
      className="container"
    >
      <input
        className="input"
        type="password"
      />
      <button
        className="icon"
        onClick={[Function]}
      >
        <Tooltip
          placement="left"
          title="Reveal"
        >
          <styled.span
            options={Object {}}
            outOfContainer={false}
            placement="left"
            type="info"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "leezwh",
                    "rules": Array [
                      "
  > :first-child:hover {
    position: relative;
  }

  > :first-child::before {
    font-family: inherit;
    visibility: hidden;
    opacity: 0;
    font-size: 0.75rem;
    line-height: 1.25;
    display: inline-block;
    box-shadow: 0px 4px 16px rgba(0, 0, 0, 0.2);
    border-radius: 4px;
    padding: 8px;
    background-color: ",
                      [Function],
                      ";
    content: attr(data-tooltip);
    color: ",
                      "#FFFFFF",
                      ";
    position: absolute;
    max-width: ",
                      [Function],
                      ";
    width: max-content;
    @supports (-ms-ime-align: auto) {
      min-width: 50px;
    }
    word-wrap: break-word;
    transform: translate(-50%, 0);
    text-align: center;
    z-index: 1001;
    transition: opacity 0.1s ease-in-out;
    ",
                      [Function],
                      "
  }

  > :first-child:hover::before {
    ",
                      [Function],
                      "
  }

  > :first-child::after {
    display: inline-block;
    visibility: hidden;
    opacity: 0;
    width: 0;
    height: 0;
    border-left: 7px solid transparent;
    border-right: 7px solid transparent;
    border-top: 7px solid
      ",
                      [Function],
                      ";
    position: absolute;
    content: '';
    z-index: 1002;
    transition: opacity 0.1s ease-in-out;
    ",
                      [Function],
                      "
  }

  > :first-child:hover::after {
    ",
                      [Function],
                      "
  }
",
                    ],
                  },
                  "displayName": "styled.span",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              options={Object {}}
              outOfContainer={false}
              placement="left"
              type="info"
            >
              <span
                className="sc-bdVaJa leezwh"
                type="info"
              >
                <div
                  data-tooltip="Reveal"
                  key=".0"
                  onMouseEnter={[Function]}
                >
                  <Component
                    hoverColor="#0E6476"
                  >
                    <Component
                      hoverColor="#0E6476"
                    >
                      <svg
                        className="svgWrapper animated"
                        style={
                          Object {
                            "--icon-active-color": "#0E353D",
                            "--icon-height": "20px",
                            "--icon-hover-color": "#0E6476",
                            "--icon-idle-color": "#0E353D",
                            "--icon-rotate": "none",
                            "--icon-width": "20px",
                          }
                        }
                        version="1.1"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clipRule="evenodd"
                          d="M10 11.5C10.8284 11.5 11.5 10.8284 11.5 10C11.5 9.17157 10.8284 8.5 10 8.5C9.17157 8.5 8.5 9.17157 8.5 10C8.5 10.8284 9.17157 11.5 10 11.5ZM10 13C11.6569 13 13 11.6569 13 10C13 8.34315 11.6569 7 10 7C8.34315 7 7 8.34315 7 10C7 11.6569 8.34315 13 10 13Z"
                          fillRule="evenodd"
                        />
                         
                        <path
                          clipRule="evenodd"
                          d="M18 10.0001C18 10.0001 16 4 10 4C4 4 2 10.0001 2 10.0001C2 10.0001 4 16 10 16C16 16 18 10.0001 18 10.0001ZM3.62005 10.0001C3.65718 10.0811 3.69973 10.1704 3.74789 10.2668C3.9914 10.7538 4.36904 11.3998 4.90232 12.0398C5.95964 13.3085 7.57163 14.5 10 14.5C12.4284 14.5 14.0404 13.3085 15.0977 12.0398C15.631 11.3998 16.0086 10.7538 16.2521 10.2668C16.3003 10.1705 16.3428 10.0811 16.38 10.0001C16.3428 9.9191 16.3003 9.82974 16.2521 9.73341C16.0086 9.24637 15.6309 8.60028 15.0977 7.96032C14.0403 6.69151 12.4283 5.5 10 5.5C7.57166 5.5 5.95967 6.69151 4.90234 7.96032C4.36905 8.60028 3.99141 9.24637 3.7479 9.73341C3.69973 9.82974 3.65718 9.9191 3.62005 10.0001Z"
                          fillRule="evenodd"
                        />
                      </svg>
                    </Component>
                  </Component>
                </div>
              </span>
            </StyledComponent>
          </styled.span>
        </Tooltip>
      </button>
    </div>
  </div>
</Component>
`;

exports[`<PasswordInput /> should render error state with feedback text 1`] = `
<Component
  feedbackText="Some feedback"
  feedbackType="error"
  hidePasswordTooltipText="Hide"
  label="Password"
  showPasswordTooltipText="Reveal"
>
  <div
    className="root"
  >
    <span
      className="label"
    >
      Password
    </span>
    <div
      className="container"
    >
      <input
        className="input error"
        type="password"
      />
      <button
        className="icon"
        onClick={[Function]}
      >
        <Tooltip
          placement="left"
          title="Reveal"
        >
          <styled.span
            options={Object {}}
            outOfContainer={false}
            placement="left"
            type="info"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "leezwh",
                    "rules": Array [
                      "
  > :first-child:hover {
    position: relative;
  }

  > :first-child::before {
    font-family: inherit;
    visibility: hidden;
    opacity: 0;
    font-size: 0.75rem;
    line-height: 1.25;
    display: inline-block;
    box-shadow: 0px 4px 16px rgba(0, 0, 0, 0.2);
    border-radius: 4px;
    padding: 8px;
    background-color: ",
                      [Function],
                      ";
    content: attr(data-tooltip);
    color: ",
                      "#FFFFFF",
                      ";
    position: absolute;
    max-width: ",
                      [Function],
                      ";
    width: max-content;
    @supports (-ms-ime-align: auto) {
      min-width: 50px;
    }
    word-wrap: break-word;
    transform: translate(-50%, 0);
    text-align: center;
    z-index: 1001;
    transition: opacity 0.1s ease-in-out;
    ",
                      [Function],
                      "
  }

  > :first-child:hover::before {
    ",
                      [Function],
                      "
  }

  > :first-child::after {
    display: inline-block;
    visibility: hidden;
    opacity: 0;
    width: 0;
    height: 0;
    border-left: 7px solid transparent;
    border-right: 7px solid transparent;
    border-top: 7px solid
      ",
                      [Function],
                      ";
    position: absolute;
    content: '';
    z-index: 1002;
    transition: opacity 0.1s ease-in-out;
    ",
                      [Function],
                      "
  }

  > :first-child:hover::after {
    ",
                      [Function],
                      "
  }
",
                    ],
                  },
                  "displayName": "styled.span",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              options={Object {}}
              outOfContainer={false}
              placement="left"
              type="info"
            >
              <span
                className="sc-bdVaJa leezwh"
                type="info"
              >
                <div
                  data-tooltip="Reveal"
                  key=".0"
                  onMouseEnter={[Function]}
                >
                  <Component
                    hoverColor="#0E6476"
                  >
                    <Component
                      hoverColor="#0E6476"
                    >
                      <svg
                        className="svgWrapper animated"
                        style={
                          Object {
                            "--icon-active-color": "#0E353D",
                            "--icon-height": "20px",
                            "--icon-hover-color": "#0E6476",
                            "--icon-idle-color": "#0E353D",
                            "--icon-rotate": "none",
                            "--icon-width": "20px",
                          }
                        }
                        version="1.1"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clipRule="evenodd"
                          d="M10 11.5C10.8284 11.5 11.5 10.8284 11.5 10C11.5 9.17157 10.8284 8.5 10 8.5C9.17157 8.5 8.5 9.17157 8.5 10C8.5 10.8284 9.17157 11.5 10 11.5ZM10 13C11.6569 13 13 11.6569 13 10C13 8.34315 11.6569 7 10 7C8.34315 7 7 8.34315 7 10C7 11.6569 8.34315 13 10 13Z"
                          fillRule="evenodd"
                        />
                         
                        <path
                          clipRule="evenodd"
                          d="M18 10.0001C18 10.0001 16 4 10 4C4 4 2 10.0001 2 10.0001C2 10.0001 4 16 10 16C16 16 18 10.0001 18 10.0001ZM3.62005 10.0001C3.65718 10.0811 3.69973 10.1704 3.74789 10.2668C3.9914 10.7538 4.36904 11.3998 4.90232 12.0398C5.95964 13.3085 7.57163 14.5 10 14.5C12.4284 14.5 14.0404 13.3085 15.0977 12.0398C15.631 11.3998 16.0086 10.7538 16.2521 10.2668C16.3003 10.1705 16.3428 10.0811 16.38 10.0001C16.3428 9.9191 16.3003 9.82974 16.2521 9.73341C16.0086 9.24637 15.6309 8.60028 15.0977 7.96032C14.0403 6.69151 12.4283 5.5 10 5.5C7.57166 5.5 5.95967 6.69151 4.90234 7.96032C4.36905 8.60028 3.99141 9.24637 3.7479 9.73341C3.69973 9.82974 3.65718 9.9191 3.62005 10.0001Z"
                          fillRule="evenodd"
                        />
                      </svg>
                    </Component>
                  </Component>
                </div>
              </span>
            </StyledComponent>
          </styled.span>
        </Tooltip>
      </button>
    </div>
    <span
      className="feedback error"
    >
      Some feedback
    </span>
  </div>
</Component>
`;

exports[`<PasswordInput /> should render error state without feedback text 1`] = `
<Component
  feedbackType="error"
  hidePasswordTooltipText="Hide"
  label="Password"
  showPasswordTooltipText="Reveal"
>
  <div
    className="root"
  >
    <span
      className="label"
    >
      Password
    </span>
    <div
      className="container"
    >
      <input
        className="input error"
        type="password"
      />
      <button
        className="icon"
        onClick={[Function]}
      >
        <Tooltip
          placement="left"
          title="Reveal"
        >
          <styled.span
            options={Object {}}
            outOfContainer={false}
            placement="left"
            type="info"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "leezwh",
                    "rules": Array [
                      "
  > :first-child:hover {
    position: relative;
  }

  > :first-child::before {
    font-family: inherit;
    visibility: hidden;
    opacity: 0;
    font-size: 0.75rem;
    line-height: 1.25;
    display: inline-block;
    box-shadow: 0px 4px 16px rgba(0, 0, 0, 0.2);
    border-radius: 4px;
    padding: 8px;
    background-color: ",
                      [Function],
                      ";
    content: attr(data-tooltip);
    color: ",
                      "#FFFFFF",
                      ";
    position: absolute;
    max-width: ",
                      [Function],
                      ";
    width: max-content;
    @supports (-ms-ime-align: auto) {
      min-width: 50px;
    }
    word-wrap: break-word;
    transform: translate(-50%, 0);
    text-align: center;
    z-index: 1001;
    transition: opacity 0.1s ease-in-out;
    ",
                      [Function],
                      "
  }

  > :first-child:hover::before {
    ",
                      [Function],
                      "
  }

  > :first-child::after {
    display: inline-block;
    visibility: hidden;
    opacity: 0;
    width: 0;
    height: 0;
    border-left: 7px solid transparent;
    border-right: 7px solid transparent;
    border-top: 7px solid
      ",
                      [Function],
                      ";
    position: absolute;
    content: '';
    z-index: 1002;
    transition: opacity 0.1s ease-in-out;
    ",
                      [Function],
                      "
  }

  > :first-child:hover::after {
    ",
                      [Function],
                      "
  }
",
                    ],
                  },
                  "displayName": "styled.span",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              options={Object {}}
              outOfContainer={false}
              placement="left"
              type="info"
            >
              <span
                className="sc-bdVaJa leezwh"
                type="info"
              >
                <div
                  data-tooltip="Reveal"
                  key=".0"
                  onMouseEnter={[Function]}
                >
                  <Component
                    hoverColor="#0E6476"
                  >
                    <Component
                      hoverColor="#0E6476"
                    >
                      <svg
                        className="svgWrapper animated"
                        style={
                          Object {
                            "--icon-active-color": "#0E353D",
                            "--icon-height": "20px",
                            "--icon-hover-color": "#0E6476",
                            "--icon-idle-color": "#0E353D",
                            "--icon-rotate": "none",
                            "--icon-width": "20px",
                          }
                        }
                        version="1.1"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clipRule="evenodd"
                          d="M10 11.5C10.8284 11.5 11.5 10.8284 11.5 10C11.5 9.17157 10.8284 8.5 10 8.5C9.17157 8.5 8.5 9.17157 8.5 10C8.5 10.8284 9.17157 11.5 10 11.5ZM10 13C11.6569 13 13 11.6569 13 10C13 8.34315 11.6569 7 10 7C8.34315 7 7 8.34315 7 10C7 11.6569 8.34315 13 10 13Z"
                          fillRule="evenodd"
                        />
                         
                        <path
                          clipRule="evenodd"
                          d="M18 10.0001C18 10.0001 16 4 10 4C4 4 2 10.0001 2 10.0001C2 10.0001 4 16 10 16C16 16 18 10.0001 18 10.0001ZM3.62005 10.0001C3.65718 10.0811 3.69973 10.1704 3.74789 10.2668C3.9914 10.7538 4.36904 11.3998 4.90232 12.0398C5.95964 13.3085 7.57163 14.5 10 14.5C12.4284 14.5 14.0404 13.3085 15.0977 12.0398C15.631 11.3998 16.0086 10.7538 16.2521 10.2668C16.3003 10.1705 16.3428 10.0811 16.38 10.0001C16.3428 9.9191 16.3003 9.82974 16.2521 9.73341C16.0086 9.24637 15.6309 8.60028 15.0977 7.96032C14.0403 6.69151 12.4283 5.5 10 5.5C7.57166 5.5 5.95967 6.69151 4.90234 7.96032C4.36905 8.60028 3.99141 9.24637 3.7479 9.73341C3.69973 9.82974 3.65718 9.9191 3.62005 10.0001Z"
                          fillRule="evenodd"
                        />
                      </svg>
                    </Component>
                  </Component>
                </div>
              </span>
            </StyledComponent>
          </styled.span>
        </Tooltip>
      </button>
    </div>
  </div>
</Component>
`;

exports[`<PasswordInput /> should render warning state with feedback text 1`] = `
<Component
  feedbackText="Some feedback"
  feedbackType="warning"
  hidePasswordTooltipText="Hide"
  label="Password"
  showPasswordTooltipText="Reveal"
>
  <div
    className="root"
  >
    <span
      className="label"
    >
      Password
    </span>
    <div
      className="container"
    >
      <input
        className="input warning"
        type="password"
      />
      <button
        className="icon"
        onClick={[Function]}
      >
        <Tooltip
          placement="left"
          title="Reveal"
        >
          <styled.span
            options={Object {}}
            outOfContainer={false}
            placement="left"
            type="info"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "leezwh",
                    "rules": Array [
                      "
  > :first-child:hover {
    position: relative;
  }

  > :first-child::before {
    font-family: inherit;
    visibility: hidden;
    opacity: 0;
    font-size: 0.75rem;
    line-height: 1.25;
    display: inline-block;
    box-shadow: 0px 4px 16px rgba(0, 0, 0, 0.2);
    border-radius: 4px;
    padding: 8px;
    background-color: ",
                      [Function],
                      ";
    content: attr(data-tooltip);
    color: ",
                      "#FFFFFF",
                      ";
    position: absolute;
    max-width: ",
                      [Function],
                      ";
    width: max-content;
    @supports (-ms-ime-align: auto) {
      min-width: 50px;
    }
    word-wrap: break-word;
    transform: translate(-50%, 0);
    text-align: center;
    z-index: 1001;
    transition: opacity 0.1s ease-in-out;
    ",
                      [Function],
                      "
  }

  > :first-child:hover::before {
    ",
                      [Function],
                      "
  }

  > :first-child::after {
    display: inline-block;
    visibility: hidden;
    opacity: 0;
    width: 0;
    height: 0;
    border-left: 7px solid transparent;
    border-right: 7px solid transparent;
    border-top: 7px solid
      ",
                      [Function],
                      ";
    position: absolute;
    content: '';
    z-index: 1002;
    transition: opacity 0.1s ease-in-out;
    ",
                      [Function],
                      "
  }

  > :first-child:hover::after {
    ",
                      [Function],
                      "
  }
",
                    ],
                  },
                  "displayName": "styled.span",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              options={Object {}}
              outOfContainer={false}
              placement="left"
              type="info"
            >
              <span
                className="sc-bdVaJa leezwh"
                type="info"
              >
                <div
                  data-tooltip="Reveal"
                  key=".0"
                  onMouseEnter={[Function]}
                >
                  <Component
                    hoverColor="#0E6476"
                  >
                    <Component
                      hoverColor="#0E6476"
                    >
                      <svg
                        className="svgWrapper animated"
                        style={
                          Object {
                            "--icon-active-color": "#0E353D",
                            "--icon-height": "20px",
                            "--icon-hover-color": "#0E6476",
                            "--icon-idle-color": "#0E353D",
                            "--icon-rotate": "none",
                            "--icon-width": "20px",
                          }
                        }
                        version="1.1"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clipRule="evenodd"
                          d="M10 11.5C10.8284 11.5 11.5 10.8284 11.5 10C11.5 9.17157 10.8284 8.5 10 8.5C9.17157 8.5 8.5 9.17157 8.5 10C8.5 10.8284 9.17157 11.5 10 11.5ZM10 13C11.6569 13 13 11.6569 13 10C13 8.34315 11.6569 7 10 7C8.34315 7 7 8.34315 7 10C7 11.6569 8.34315 13 10 13Z"
                          fillRule="evenodd"
                        />
                         
                        <path
                          clipRule="evenodd"
                          d="M18 10.0001C18 10.0001 16 4 10 4C4 4 2 10.0001 2 10.0001C2 10.0001 4 16 10 16C16 16 18 10.0001 18 10.0001ZM3.62005 10.0001C3.65718 10.0811 3.69973 10.1704 3.74789 10.2668C3.9914 10.7538 4.36904 11.3998 4.90232 12.0398C5.95964 13.3085 7.57163 14.5 10 14.5C12.4284 14.5 14.0404 13.3085 15.0977 12.0398C15.631 11.3998 16.0086 10.7538 16.2521 10.2668C16.3003 10.1705 16.3428 10.0811 16.38 10.0001C16.3428 9.9191 16.3003 9.82974 16.2521 9.73341C16.0086 9.24637 15.6309 8.60028 15.0977 7.96032C14.0403 6.69151 12.4283 5.5 10 5.5C7.57166 5.5 5.95967 6.69151 4.90234 7.96032C4.36905 8.60028 3.99141 9.24637 3.7479 9.73341C3.69973 9.82974 3.65718 9.9191 3.62005 10.0001Z"
                          fillRule="evenodd"
                        />
                      </svg>
                    </Component>
                  </Component>
                </div>
              </span>
            </StyledComponent>
          </styled.span>
        </Tooltip>
      </button>
    </div>
    <span
      className="feedback warning"
    >
      Some feedback
    </span>
  </div>
</Component>
`;

exports[`<PasswordInput /> should render warning state without feedback text 1`] = `
<Component
  feedbackType="warning"
  hidePasswordTooltipText="Hide"
  label="Password"
  showPasswordTooltipText="Reveal"
>
  <div
    className="root"
  >
    <span
      className="label"
    >
      Password
    </span>
    <div
      className="container"
    >
      <input
        className="input warning"
        type="password"
      />
      <button
        className="icon"
        onClick={[Function]}
      >
        <Tooltip
          placement="left"
          title="Reveal"
        >
          <styled.span
            options={Object {}}
            outOfContainer={false}
            placement="left"
            type="info"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "leezwh",
                    "rules": Array [
                      "
  > :first-child:hover {
    position: relative;
  }

  > :first-child::before {
    font-family: inherit;
    visibility: hidden;
    opacity: 0;
    font-size: 0.75rem;
    line-height: 1.25;
    display: inline-block;
    box-shadow: 0px 4px 16px rgba(0, 0, 0, 0.2);
    border-radius: 4px;
    padding: 8px;
    background-color: ",
                      [Function],
                      ";
    content: attr(data-tooltip);
    color: ",
                      "#FFFFFF",
                      ";
    position: absolute;
    max-width: ",
                      [Function],
                      ";
    width: max-content;
    @supports (-ms-ime-align: auto) {
      min-width: 50px;
    }
    word-wrap: break-word;
    transform: translate(-50%, 0);
    text-align: center;
    z-index: 1001;
    transition: opacity 0.1s ease-in-out;
    ",
                      [Function],
                      "
  }

  > :first-child:hover::before {
    ",
                      [Function],
                      "
  }

  > :first-child::after {
    display: inline-block;
    visibility: hidden;
    opacity: 0;
    width: 0;
    height: 0;
    border-left: 7px solid transparent;
    border-right: 7px solid transparent;
    border-top: 7px solid
      ",
                      [Function],
                      ";
    position: absolute;
    content: '';
    z-index: 1002;
    transition: opacity 0.1s ease-in-out;
    ",
                      [Function],
                      "
  }

  > :first-child:hover::after {
    ",
                      [Function],
                      "
  }
",
                    ],
                  },
                  "displayName": "styled.span",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              options={Object {}}
              outOfContainer={false}
              placement="left"
              type="info"
            >
              <span
                className="sc-bdVaJa leezwh"
                type="info"
              >
                <div
                  data-tooltip="Reveal"
                  key=".0"
                  onMouseEnter={[Function]}
                >
                  <Component
                    hoverColor="#0E6476"
                  >
                    <Component
                      hoverColor="#0E6476"
                    >
                      <svg
                        className="svgWrapper animated"
                        style={
                          Object {
                            "--icon-active-color": "#0E353D",
                            "--icon-height": "20px",
                            "--icon-hover-color": "#0E6476",
                            "--icon-idle-color": "#0E353D",
                            "--icon-rotate": "none",
                            "--icon-width": "20px",
                          }
                        }
                        version="1.1"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clipRule="evenodd"
                          d="M10 11.5C10.8284 11.5 11.5 10.8284 11.5 10C11.5 9.17157 10.8284 8.5 10 8.5C9.17157 8.5 8.5 9.17157 8.5 10C8.5 10.8284 9.17157 11.5 10 11.5ZM10 13C11.6569 13 13 11.6569 13 10C13 8.34315 11.6569 7 10 7C8.34315 7 7 8.34315 7 10C7 11.6569 8.34315 13 10 13Z"
                          fillRule="evenodd"
                        />
                         
                        <path
                          clipRule="evenodd"
                          d="M18 10.0001C18 10.0001 16 4 10 4C4 4 2 10.0001 2 10.0001C2 10.0001 4 16 10 16C16 16 18 10.0001 18 10.0001ZM3.62005 10.0001C3.65718 10.0811 3.69973 10.1704 3.74789 10.2668C3.9914 10.7538 4.36904 11.3998 4.90232 12.0398C5.95964 13.3085 7.57163 14.5 10 14.5C12.4284 14.5 14.0404 13.3085 15.0977 12.0398C15.631 11.3998 16.0086 10.7538 16.2521 10.2668C16.3003 10.1705 16.3428 10.0811 16.38 10.0001C16.3428 9.9191 16.3003 9.82974 16.2521 9.73341C16.0086 9.24637 15.6309 8.60028 15.0977 7.96032C14.0403 6.69151 12.4283 5.5 10 5.5C7.57166 5.5 5.95967 6.69151 4.90234 7.96032C4.36905 8.60028 3.99141 9.24637 3.7479 9.73341C3.69973 9.82974 3.65718 9.9191 3.62005 10.0001Z"
                          fillRule="evenodd"
                        />
                      </svg>
                    </Component>
                  </Component>
                </div>
              </span>
            </StyledComponent>
          </styled.span>
        </Tooltip>
      </button>
    </div>
  </div>
</Component>
`;
